src/Cake.CodeAnalysisReporting/Reports/msbuild-xmlfilelogger-by-assembly.xsl

```diff
@@ -0,0 +1,512 @@
+        background-repeat: no-repeat;
+        background-position-y: center;
+        }
+
+        #msbuild-report .inner-results td.warning
+        {
+        background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAABGdBTUEAAK/INwWK6QAAABl0RVh0U29mdHdhcmUAQWRvYmUgSW1hZ2VSZWFkeXHJZTwAAAIsSURBVDjLpVNLSJQBEP7+h6uu62vLVAJDW1KQTMrINQ1vPQzq1GOpa9EppGOHLh0kCEKL7JBEhVCHihAsESyJiE4FWShGRmauu7KYiv6Pma+DGoFrBQ7MzGFmPr5vmDFIYj1mr1WYfrHPovA9VVOqbC7e/1rS9ZlrAVDYHig5WB0oPtBI0TNrUiC5yhP9jeF4X8NPcWfopoY48XT39PjjXeF0vWkZqOjd7LJYrmGasHPCCJbHwhS9/F8M4s8baid764Xi0Ilfp5voorpJfn2wwx/r3l77TwZUvR+qajXVn8PnvocYfXYH6k2ioOaCpaIdf11ivDcayyiMVudsOYqFb60gARJYHG9DbqQFmSVNjaO3K2NpAeK90ZCqtgcrjkP9aUCXp0moetDFEeRXnYCKXhm+uTW0CkBFu4JlxzZkFlbASz4CQGQVBFeEwZm8geyiMuRVntzsL3oXV+YMkvjRsydC1U+lhwZsWXgHb+oWVAEzIwvzyVlk5igsi7DymmHlHsFQR50rjl+981Jy1Fw6Gu0ObTtnU+cgs28AKgDiy+Awpj5OACBAhZ/qh2HOo6i+NeA73jUAML4/qWux8mt6NjW1w599CS9xb0mSEqQBEDAtwqALUmBaG5FV3oYPnTHMjAwetlWksyByaukxQg2wQ9FlccaK/OXA3/uAEUDp3rNIDQ1ctSk6kHh1/jRFoaL4M4snEMeD73gQx4M4PsT1IZ5AfYH68tZY7zv/ApRMY9mnuVMvAAAAAElFTkSuQmCC');
+        background-repeat: no-repeat;
+        background-position-y: center;
+        }
+
+        #msbuild-report .inner-header th
+        {
+        background: #b9c9fe;
+        color: #039;
+        }
+
+        #msbuild-report .inner-rule-description
+        {
+        background-color: transparent;
+        border-collapse: collapse;
+        border: 0px;
+        font-size: 12px;
+        margin-bottom: 3px;
+        margin-top: 4px;
+        text-align: left;
+        width: 100%;
+        }
+
+        #msbuild-report .inner-rule-description tr
+        {
+        background-color: transparent;
+        border: 0px;
+        }
+
+        #msbuild-report .inner-rule-description td
+        {
+        background-color: transparent;
+        border: 0px;
+        }
+      </style>
+      <xsl:apply-templates select="$msbuild.root" />
+    </div>
+  </xsl:template>
+
+  <xsl:template match="build">
+    <!-- Remove .sln file from path to get the path of the solution file -->
```

Comment:
This assumes a specific directory structure. Didn't had any idea for a more generic approach though